connectivity/service/src/main/java/org/eclipse/ditto/connectivity/service/messaging/amqp/AmqpClientActor.java

```diff
@@ -397,32 +396,6 @@ protected void cleanupFurtherResourcesOnConnectionTimeout(final BaseClientState
         super.cleanupFurtherResourcesOnConnectionTimeout(currentState);
     }
 
-    @Override
-    protected boolean isEventUpToDate(final Object event, final BaseClientState state,
-            @Nullable final ActorRef sender) {
-        if (getSelf().equals(sender)) {
-            // events from self are always relevant
-            return true;
-        }
-        switch (state) {
-            case CONNECTED:
-                // while connected, events from publisher or consumer child actors are relevant
-                return sender != null &&
-                        sender.path().toStringWithoutAddress().startsWith(getSelf().path().toStringWithoutAddress()) &&
-                        (sender.path().name().startsWith(AmqpConsumerActor.ACTOR_NAME_PREFIX) ||
-                                sender.path().name().startsWith(AmqpPublisherActor.ACTOR_NAME_PREFIX));
-            case CONNECTING:
-                return Objects.equals(sender, connectConnectionHandler);
-            case DISCONNECTING:
-                return Objects.equals(sender, disconnectConnectionHandler);
-            case TESTING:
-            default:
-                // no need to check testConnectionHandler because test runs only once during this actor's lifetime
-                // ignore random events by default - they could come from a connection handler that is already dead
-                return false;
-        }
-    }
```

Comment:
@yufei-cai why did you remove this? 
If it's not required, it's sure a good thing, but I don't understand how you came to the conclusion to remove it :)

Reply:
The reason is that connection failures can now come from consumer actors in connected state. Instead of maintaining the "isEventUpToDate"  table for this change and all future changes I think it better to risk restarting due to stale events than to ignore relevant events.